### ISSUE TITLE
Shift-Multi-Selection

### ISSUE THREAD

**Issue by Rastopapola** [sebastianpauli/netgis-client#29]
# Status quo
Bspw. beim `Löschen` wird eine Geometrie gelöscht, dann wechselt das Tool zurück auf `Betrachten`. Diese Änderung kam aufgrund von Nutzern, die immer weiter Geometrien löschten, obwohl sie das nicht wollten.

# Enhancement
Ein Nutzer regte an, bei gedrückter `Shift` Taste mehrere Objekte markieren zu können, auf die das Tool dann angewendet wird. Das ließe sich auch bspw. beim `Puffern` nutzen. Ein Beispielablauf:

* Nutzer wählt `Löschen`
* Nutzer hält Shift gedrückt
    * Nutzer klickt mehrere Objekte an
    * Gewählte Objekte werden blau hervorgehoben
    * Nutzer hält Shift nicht mehr gedrückt
* Das gewählte Tool wird auf die markierten Objekte angewendet (hier --> `Löschen`)

**Comment by sebastianpauli** [comment]
Added in latest [LANIS Update](https://github.com/sebastianpauli/netgis-client/commit/81fa3bef489b589c9e8252277923ce7012e07093).

You can now hold Shift + Click to add features to a tool selection (cut, buffer).

**Comment by Rastopapola** [comment]
Not working for me in most recent version (https://github.com/sebastianpauli/netgis-client/commit/bb4bc31d8a915176848856014e29297fd585f2cb). 
Activating the tool, then holding shift and select the features I want to be processed does not work. There is still only the first clicked being processed.

**Comment by sebastianpauli** [comment]
Should be fixed in [latest commit](https://github.com/sebastianpauli/netgis-client/commit/e9fee75f61f9b9a708be80b480f33447e68f2316).

Rewrote the buffer/cut tools selection code to be more dynamic. While either the buffer or cut tool is activated you can now add or remove features to the selection using these inputs:

- Click (without Shift) on a feature to make this the single selection
- Shift + Click on a feature to add it to the selection (the buffer preview should automatically update, should also work while drawing the cut polygon by temporarily going back to select mode)
- Click (without Shift) on empty space to clear the selection

**Comment by Rastopapola** [comment]
>Shift + Click on a feature to add it to the selection (the buffer preview should automatically update, should also work while drawing the cut polygon by temporarily going back to select mode)

At first I struggled trying to get this to work on my machine but then I realized you implemented 

SHIFT + Click on Polygon --> adds to buffer

Release SHIFT

Again press SHIFT + Click on Polygon --> adds to buffer

However, I was surprsied it is not 

SHIFT (press) + click --> add to buffer + click --> add to buffer + SHIFT (release) --> end buffer pushing

I am not sure wheter regular users would understand shift-multi selection any other way?

**Comment by sebastianpauli** [comment]
Just added a new config option in the [latest commit](https://github.com/sebastianpauli/netgis-client/commit/1df99be964169ccbe40d77421e5f3afc5997292c).

In config[ "tools" ] you can now set:
```
"select_multi_reset": true
```

This will then reset the feature selection on the next click after the SHIFT key has been released. Setting this to false will revert back to the previous behavior where the user keeps adding to the selection as long as the SHIFT key is pressed down.

**Comment by sebastianpauli** [comment]
Tested the behavior on Debian + Firefox but could not reproduce the bug (holding Shift and Clicking to add features works like in e.g. QGIS for me).

But to work around this problem and to make multi select work on systems without keyboard, I implemented an option for the Buffer Tool to **toggle multi select from the GUI** in the [latest commit](https://github.com/sebastianpauli/netgis-client/commit/4c54e79feddf545859f713ffac1372a19b48a34c):

![Image](https://github.com/user-attachments/assets/6c1e1172-81e6-4cf0-9069-687a293fa4be)

Hope this helps!
